### PR TITLE
Updated forceMD5Etag to boolean pointer type

### PR DESCRIPTION
**Pull request by achouhan09**
### Explain the changes
1. Updated forceMd5Etag to a Boolean Pointer Type (*bool): The forceMd5Etag flag is now handled as a pointer to a boolean (*bool) to explicitly track whether the flag was provided by the user:

- If the flag was set, forceMd5Etag points to a true or false value. 
- If the flag was not set, forceMd5Etag remains nil, preventing unintended changes to the ForceMd5Etag field.

### Issues: Fixed #xxx / Gap #xxx
1. Improvement: #1215 
2. BZ: https://bugzilla.redhat.com/show_bug.cgi?id=2312533

### Testing Instructions:
1. Try creating and updating the bucket using cli.
2. Try creating and updating the NoobaaAccount using cli and yaml both.

### CODE REVIEW

**Comment by jackyalbo** [comment]
@achouhan09 I remember @dannyzaken had issues when changing API schemas between versions. Did you test upgrade where there was a noobaaAccount with true/false set and if it worked after moving to the new schema in the later version? 
Try doing a whole upgrade procedure of running with the old CRD and then changing to the new one and see if everything still works(can maybe check with the cli upgrade command - if it supports upgrading the CRDs)

File: pkg/apis/noobaa/v1alpha1/noobaaaccount_types.go [hunk 1]
```diff
@@ -69,7 +69,7 @@ type NooBaaAccountSpec struct {
 
 	// ForceMd5Etag specifies whether MD5 Etags should be calculated for the account or not
 	// +optional
-	ForceMd5Etag *bool `json:"force_md5_etag,omitempty"`
+	ForceMd5Etag bool `json:"force_md5_etag,omitempty"`
```
Comment (dannyzaken) [inline pkg/apis/noobaa/v1alpha1/noobaaaccount_types.go:72]: Why is this change necessary? 
With this code change, the user must set a value for `ForceMd5Etag`. If the user leaves this field empty, it will implicitly be set to `false` and override any other setting\default in noobaa-core.

Comment (achouhan09) [reply pkg/apis/noobaa/v1alpha1/noobaaaccount_types.go]: Done the required changes as per discussed in slack, everything working as expected. Thanks

**Review by dannyzaken — CHANGES_REQUESTED** [review]

File: pkg/noobaaaccount/noobaaaccount.go [hunk 2]
```diff
@@ -272,11 +272,7 @@ func RunUpdate(cmd *cobra.Command, args []string) {
 	name := args[0]
 
 	newDefaultResource := util.GetFlagStringOrPrompt(cmd, "new_default_resource")
-	var forceMd5EtagPtr *bool = nil
-	if cmd.Flags().Changed("force_md5_etag") {
-		forceMd5Etag, _ := cmd.Flags().GetBool("force_md5_etag")
-		forceMd5EtagPtr = &forceMd5Etag
-	}
+	forceMd5Etag, _ := cmd.Flags().GetBool("force_md5_etag")
```
[the remark this answers is not in the document]
Comment (achouhan09) [reply pkg/noobaaaccount/noobaaaccount.go]: Make sense, Thanks.

File: pkg/bucket/bucket.go [hunk 3]
```diff
@@ -187,7 +194,7 @@ func RunStatus(cmd *cobra.Command, args []string) {
 	}
 	fmt.Printf("  %-22s : %s\n", "Type", b.BucketType)
 	fmt.Printf("  %-22s : %s\n", "Mode", b.Mode)
-	fmt.Printf("  %-22s : %t\n", "Force Md5 Etag", b.ForceMd5Etag)
+	fmt.Printf("  %-22s : %t\n", "Force Md5 Etag", *b.ForceMd5Etag)
```
[the remark this answers is not in the document]
Comment (achouhan09) [reply pkg/bucket/bucket.go]: Done, Thanks

File: pkg/util/util.go [hunk 4]
```diff
@@ -1238,6 +1238,20 @@ func GetFlagStringOrPromptPassword(cmd *cobra.Command, flag string) string {
 	return str
 }
 
+// GetBoolFlagPtr returns a pointer to the boolean flag value if set, or nil if not set
+func GetBoolFlagPtr(cmd *cobra.Command, flag string) (*bool, error) {
+	flagVal, err := cmd.Flags().GetBool(flag)
+	if err != nil {
+		return nil, err
+	}
```
Comment (achouhan09) [inline pkg/util/util.go:1246]: If I am adding this condition, the admission-tests are failing with the below error:
`time="2024-09-20T12:55:53Z" level=fatal msg="flag accessed but not defined: force_md5_etag"`

I think we need not to handle this condition😅 just like other functions in util.go, also without it the tests are passing. As we are returning nil value from the function if the flag is not found.
@dannyzaken  wdyt?

Comment (dannyzaken) [reply pkg/util/util.go]: What is the root cause of this error? I am not sure ignoring the error is the answer

File: pkg/util/util.go [hunk 5]
```diff
@@ -1238,6 +1238,17 @@ func GetFlagStringOrPromptPassword(cmd *cobra.Command, flag string) string {
 	return str
 }
 
+// GetBoolFlagPtr returns a pointer to the boolean flag value if set, or nil if not set
+func GetBoolFlagPtr(cmd *cobra.Command, flag string) *bool {
+	flagVal, _ := cmd.Flags().GetBool(flag)
```
Comment (dannyzaken) [inline pkg/util/util.go:1243]: This should be inside the if. I think this is the cause for the tests failing

File: pkg/util/util.go [hunk 6]
```diff
@@ -1238,6 +1238,17 @@ func GetFlagStringOrPromptPassword(cmd *cobra.Command, flag string) string {
...
+
+	if cmd.Flags().Changed(flag) {
+		return &flagVal
+	}
+
+	return nil
+}
```
Comment (dannyzaken) [inline pkg/util/util.go:1250]: The `GetBool` should be inside the `if` statement. I think this is why the tests failed. 
Also, we should not ignore the error. Either return it or handle it in another way (log, etc.).

```suggestion
	if cmd.Flags().Changed(flag) {
	        flagVal, err := cmd.Flags().GetBool(flag)
        	if err != nil {
        	// handle error
 	        }
        	return &flagVal
	}
	return nil
}
```

**Review by dannyzaken — APPROVED** [review]
